### PR TITLE
[SPARK-42494] Add official image Dockerfile for Spark v3.3.2

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?
Add Apache Spark 3.3.2 Dockerfiles.
- Add 3.3.2 GPG key
- Add .github/workflows/build_3.3.2.yaml
- ./add-dockerfiles.sh 3.3.2

### Why are the changes needed?
Apache Spark 3.3.2 released.

https://lists.apache.org/thread/k8skf16wyn6rg9n0vd0t6l3bhw7c9svq

### Does this PR introduce _any_ user-facing change?
Yes in future, new image will publised in future (after DOI reviewed)

### How was this patch tested?
Add workflow and CI passed

### CODE REVIEW

**Comment by Yikun** [comment]
cc @viirya @HyukjinKwon

File: 3.3.2/scala2.12-java11-r-ubuntu/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+if [ -n "$SPARK_EXTRA_CLASSPATH" ]; then
+  SPARK_CLASSPATH="$SPARK_CLASSPATH:$SPARK_EXTRA_CLASSPATH"
+fi
+
+if ! [ -z ${PYSPARK_PYTHON+x} ]; then
+    export PYSPARK_PYTHON
+fi
+if ! [ -z ${PYSPARK_DRIVER_PYTHON+x} ]; then
+    export PYSPARK_DRIVER_PYTHON
+fi
```
Comment (viirya) [inline 3.3.2/scala2.12-java11-r-ubuntu/entrypoint.sh:53]: Do we need PySpark vars for R docker file?

File: 3.3.2/scala2.12-java11-ubuntu/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+if [ -n "$SPARK_EXTRA_CLASSPATH" ]; then
+  SPARK_CLASSPATH="$SPARK_CLASSPATH:$SPARK_EXTRA_CLASSPATH"
+fi
+
+if ! [ -z ${PYSPARK_PYTHON+x} ]; then
+    export PYSPARK_PYTHON
+fi
+if ! [ -z ${PYSPARK_DRIVER_PYTHON+x} ]; then
+    export PYSPARK_DRIVER_PYTHON
+fi
```
Comment (viirya) [inline 3.3.2/scala2.12-java11-ubuntu/entrypoint.sh:53]: ditto

**Review by viirya — APPROVED** [review]

File: 3.3.2/scala2.12-java11-r-ubuntu/entrypoint.sh (hunk 1 above)
Comment (Yikun) [reply 3.3.2/scala2.12-java11-r-ubuntu/entrypoint.sh:53]: Good point, I also check the original changes [1], it's for PySpark only.

I will get this merged first (3.2.2 dockerfiles). And submit a new PR to support entrypoint.template

[1] https://github.com/apache/spark/commit/a99a47ca1df689377dbfbf4dd7258f59aee2be44

**Comment by Yikun** [comment]
@viirya @HyukjinKwon Merged, thanks!

**Pull request closed without merging**